ui/src/store/device-control-store.ts

```diff
@@ -19,8 +19,40 @@ class DeviceControlStore {
     this.currentQuality = quality
   }
 
+  goHome(serial: string): void {
+    const { data: device } = deviceBySerialStore.deviceQueryResult(serial)
+
+    if (!device?.channel) return
+
+    controlService.home(device.channel)
+  }
+
+  openMenu(serial: string): void {
+    const { data: device } = deviceBySerialStore.deviceQueryResult(serial)
+
+    if (!device?.channel) return
+
+    controlService.menu(device.channel)
```

Comment:
This looks repeating. Can we extract a common function or use a fabric pattern to get rid of the repeating code?

Maybe use
```suggestion
    pressButton("menu")
```
and have
```javascript
const pressButton = (buttonName: str /* or an enum for that matter */ ) => {
  const { data: device } = deviceBySerialStore.deviceQueryResult(serial)

    if (!device?.channel) return

    controlService[buttonName](device.channel)
}
```


But I don't like this either

Why do we need such a method separately?